internal/advertisement-operator/controller.go

```diff
@@ -93,10 +100,10 @@ func (r *AdvertisementReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
 			return ctrl.Result{}, err
 		}
 		klog.V(3).Info("Correct creation of virtual kubelet deployment for cluster " + adv.Spec.ClusterId)
-		return ctrl.Result{}, nil
+		return ctrl.Result{RequeueAfter: r.RetryTimeout}, nil
 	}
 
-	return ctrl.Result{}, nil
+	return ctrl.Result{RequeueAfter: r.RetryTimeout}, nil
```

Comment:
when we arrive here, it means everything went well in reconciliation: do we need/want to reque nonetheless?

Reply:
I think you are right. This is not necessary if we correctly set up the owner reference on created objects.